modules/agent-policy/scripts/script-utils.sh

```diff
@@ -0,0 +1,332 @@
+#!/usr/bin/env bash
+
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+####################################################################
+## This script contains util functions. The util functions are used
+## in modules/agent-policy/scripts/create-update-script.sh and
+## modules/agent-policy/scripts/delete-script.sh
+####################################################################
+
+
+CREATE="create"
+UPDATE="update"
+LAUNCH_STAGE="alpha"
+
+
+# Params:
+#   $1 = JSON formatted list(string)
+# Return:
+#   A well-formatted command line flag value for a list of strings
+function get_formatted_list_of_strings() {
+    local formatted
+    local python="python -c 'import json, sys;"
+    python="$python list_of_strings = json.load(sys.stdin);"
+    python="$python print (\",\".join(x for x in list_of_strings))'"
+    formatted="$(echo "$1" | eval "$python")"
+    echo "$formatted"
+}
+
+
+# Params:
+#   $1 = JSON formatted list(object)
+# Return:
+#   A well-formatted command line flag value for a list of objects
+function get_formatted_list_of_objects() {
+    local formatted
+    local python="python -c 'import json, sys;"
+    python="$python list_of_objs = json.load(sys.stdin);"
+    python="$python print (\";\".join(\",\".join([\"{}={}\".format(k.replace(\"_\", \"-\"),"
+    python="$python str(v).lower() if type(v) is bool else v) for k, v in obj.items()])"
+    python="$python for obj in list_of_objs))'"
+    formatted="$(echo "$1" | eval "$python")"
+    echo "$formatted"
+}
+
+
+# Params:
+#   $1 = JSON formatted list(list(objects))
+# Return:
+#   A well-formatted command line flag value for a list of list of objects
+function get_formatted_list_of_list_of_objects() {
+    local formatted
+    local python="python -c 'import json, sys;"
+    python="$python list_of_list_of_objs = json.load(sys.stdin);"
+    python="$python print (\";\".join(\",\".join(\"=\".join(inner_list[x] for x in inner_list)"
+    python="$python for inner_list in outer_list) for outer_list in list_of_list_of_objs))'"
+    formatted="$(echo "$1" | eval "$python")"
+    echo "$formatted"
+
+}
+
+# Params:
+#   $1 = output of successful describe command
+# Return:
+#   the etag in the given string
+function get_etag() {
+    local describe_output="$1"
+    local etag=${describe_output#*etag: } # removes everything before etag
```

Comment:
This logic could be less brittle with JSON output + jq, right now we're very vulnerable to subtle output changes.

Reply:
Will do separate pr for this.

Reply:
https://github.com/terraform-google-modules/terraform-google-cloud-operations/pull/7